GPlay bug report IllegalArgumentException: the bind value at index 1 is null

Seems to be a Null exception again.
```
java.lang.RuntimeException: An error occured while executing doInBackground()
	at android.os.AsyncTask$3.done(AsyncTask.java:200)
	at java.util.concurrent.FutureTask$Sync.innerSetException(FutureTask.java:274)
	at java.util.concurrent.FutureTask.setException(FutureTask.java:125)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:308)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1088)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:581)
	at java.lang.Thread.run(Thread.java:1019)
Caused by: java.lang.IllegalArgumentException: the bind value at index 1 is null
	at android.database.sqlite.SQLiteProgram.bindString(SQLiteProgram.java:237)
	at android.database.sqlite.SQLiteQuery.bindString(SQLiteQuery.java:185)
	at android.database.sqlite.SQLiteDirectCursorDriver.query(SQLiteDirectCursorDriver.java:48)
	at android.database.sqlite.SQLiteDatabase.rawQueryWithFactory(SQLiteDatabase.java:1358)
	at android.database.sqlite.SQLiteDatabase.rawQuery(SQLiteDatabase.java:1326)
	at de.greenrobot.dao.query.Query.unique(Query.java:170)
	at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.processFeed(UpdateFeedTask.java:247)
	at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.updateByFeed(UpdateFeedTask.java:169)
	at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.updateAllFeeds(UpdateFeedTask.java:103)
	at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.doInBackground(UpdateFeedTask.java:67)
	at fr.gaulupeau.apps.Poche.network.tasks.UpdateFeedTask.doInBackground(UpdateFeedTask.java:30)
	at android.os.AsyncTask$2.call(AsyncTask.java:185)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:306)
	... 4 more
```

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -39,7 +39,7 @@ dependencies {
     compile 'com.android.support:recyclerview-v7:23.1.1'
     compile 'com.android.support:design:23.1.1'
     compile 'de.greenrobot:greendao:2.0.0'
+    compile 'com.squareup.okhttp:okhttp:2.7.0'
     compile 'com.facebook.stetho:stetho:1.2.0'
-    compile 'com.squareup.okhttp:okhttp:2.5.0'
     compile 'com.facebook.stetho:stetho-okhttp:1.2.0'
 }

--- a/app/src/main/java/fr/gaulupeau/apps/Poche/network/tasks/UpdateFeedTask.java
+++ b/app/src/main/java/fr/gaulupeau/apps/Poche/network/tasks/UpdateFeedTask.java
@@ -214,6 +214,8 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
     private void processFeed(ArticleDao articleDao, InputStream is,
                              FeedType feedType, UpdateType updateType, Integer latestID)
             throws XmlPullParserException, IOException {
+        // TODO: use parser.require() all over the place?
+
         XmlPullParser parser = Xml.newPullParser();
         parser.setFeature(XmlPullParser.FEATURE_PROCESS_NAMESPACES, false);
         parser.setInput(is, null);
@@ -239,6 +241,7 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
                     // Favorite: Full
 
                     Item item = parseItem(parser);
+
                     Integer id = getIDFromURL(item.sourceUrl);
 
                     if(updateType == UpdateType.Fast && latestID != null && id != null
@@ -336,21 +339,21 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
 
             switch (parser.getName()) {
                 case "title":
-                    item.title = cleanString(parser.nextText());
+                    item.title = cleanString(parseNextText(parser));
                     break;
                 case "source":
                     String sourceUrl = parser.getAttributeValue(null, "url");
-                    parser.nextText(); // ignore "empty" element
+                    parseNextText(parser); // ignore "empty" element
                     item.sourceUrl = sourceUrl;
                     break;
                 case "link":
-                    item.link = parser.nextText();
+                    item.link = parseNextText(parser);
                     break;
                 case "pubDate":
-                    item.pubDate = parser.nextText();
+                    item.pubDate = parseNextText(parser);
                     break;
                 case "description":
-                    item.description = parser.nextText();
+                    item.description = parseNextText(parser);
                     break;
 
                 default:
@@ -360,6 +363,16 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
         }
 
         return item;
+    }
+
+    private static String parseNextText(XmlPullParser parser)
+            throws XmlPullParserException, IOException {
+        String result = parser.nextText();
+
+        // workaround for Android 2.3.3 XmlPullParser.nextText() bug
+        if(parser.getEventType() != XmlPullParser.END_TAG) parser.next();
+
+        return result;
     }
 
     private static Integer parseItemID(XmlPullParser parser) throws XmlPullParserException, IOException {

--- a/app/src/main/java/fr/gaulupeau/apps/Poche/network/tasks/UpdateFeedTask.java
+++ b/app/src/main/java/fr/gaulupeau/apps/Poche/network/tasks/UpdateFeedTask.java
@@ -41,6 +41,8 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
         }
     }
 
+    private static final String TAG = UpdateFeedTask.class.getSimpleName();
+
     private String baseURL;
     private String apiUserId;
     private String apiToken;
@@ -93,29 +95,43 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
     }
 
     private void updateAllFeeds() {
+        Log.i(TAG, "updateAllFeeds() started");
+
         ArticleDao articleDao = DbConnection.getSession().getArticleDao();
         SQLiteDatabase db = articleDao.getDatabase();
 
+        Log.d(TAG, "updateAllFeeds() beginning transaction");
         db.beginTransaction();
         try {
+            Log.d(TAG, "updateAllFeeds() deleting old articles");
             articleDao.deleteAll();
 
+            Log.d(TAG, "updateAllFeeds() updating Main feed");
             if(!updateByFeed(articleDao, FeedType.Main, UpdateType.Full, 0)) {
+                Log.w(TAG, "updateAllFeeds() Main feed update failed; exiting");
                 return;
             }
 
+            Log.d(TAG, "updateAllFeeds() updating Archive feed");
             if(!updateByFeed(articleDao, FeedType.Archive, UpdateType.Full, 0)) {
+                Log.w(TAG, "updateAllFeeds() Archive feed update failed; exiting");
                 return;
             }
 
+            Log.d(TAG, "updateAllFeeds() updating Favorite feed");
             if(!updateByFeed(articleDao, FeedType.Favorite, UpdateType.Fast, 0)) {
+                Log.w(TAG, "updateAllFeeds() Favorite feed update failed; exiting");
                 return;
             }
 
+            Log.d(TAG, "updateAllFeeds() setting transaction successful");
             db.setTransactionSuccessful();
         } finally {
+            Log.d(TAG, "updateAllFeeds() ending transaction");
             db.endTransaction();
         }
+
+        Log.d(TAG, "updateAllFeeds() finished");
     }
 
     private void update(FeedType feedType, UpdateType updateType) {
@@ -150,6 +166,8 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
 
     private boolean updateByFeed(ArticleDao articleDao, FeedType feedType, UpdateType updateType,
                                  Integer id) {
+        Log.d(TAG, "updateByFeed() started");
+
         InputStream is = null;
         try {
             // TODO: rewrite?
@@ -165,6 +183,7 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
                 return false;
             }
 
+            Log.d(TAG, "updateByFeed() got input stream; processing feed");
             try {
                 processFeed(articleDao, is, feedType, updateType, id);
             } catch (IOException e) {
@@ -177,6 +196,8 @@ public class UpdateFeedTask extends AsyncTask<Void, Void, Void> {
                 errorMessage = App.getInstance().getString(R.string.feedUpdater_feedProcessingError);
                 return false;
             }
+
+            Log.d(TAG, "updateByFeed() finished successfully");
 
             return true;
         } finally {

